File: extract_json.dart
// ignore_for_file: avoid_print
import 'dart:convert';
import 'dart:io';

void main() async {
  try {
    final file = File('scraper_html.html');
    if (!await file.exists()) {
      print('scraper_html.html not found');
      return;
    }
    
    final html = await file.readAsString();
    
    final startMarker = 'window.APP_INITIALIZATION_STATE=';
    final endMarker = ';window.APP_FLAGS=';
    
    final startIndex = html.indexOf(startMarker);
    if (startIndex == -1) {
      print('APP_INITIALIZATION_STATE not found');
      return;
    }
    
    final dataStartIndex = startIndex + startMarker.length;
    final endIndex = html.indexOf(endMarker, dataStartIndex);
    
    if (endIndex == -1) {
      print('APP_FLAGS not found after APP_INITIALIZATION_STATE');
      return;
    }
    
    final jsonString = html.substring(dataStartIndex, endIndex);
    final data = jsonDecode(jsonString);
    
    print('Data type: ${data.runtimeType}');
    if (data is List) {
      print('List length: ${data.length}');
      for (var i = 0; i < data.length; i++) {
        // print summary of each top level element
        final element = data[i];
        if (element is List) {
           print('Item $i is List of length ${element.length}');
        } else {
           print('Item $i is ${element.runtimeType}');
        }
      }
    }
    
    // Write formatted JSON to file
    final encoder = JsonEncoder.withIndent('  ');
    final formattedJson = encoder.convert(data);
    
    await File('extracted_data.json').writeAsString(formattedJson);
    print('Successfully wrote formatted JSON to extracted_data.json');
    
  } catch (e) {
    print('Error: $e');
  }
}

File: extract_json.py
import json
import re
import sys

def main():
    try:
        with open('scraper_html.html', 'r', encoding='utf-8') as f:
            html = f.read()
        
        # Look for window.APP_INITIALIZATION_STATE=
        start_marker = 'window.APP_INITIALIZATION_STATE='
        end_marker = ';window.APP_FLAGS='
        
        start_idx = html.find(start_marker)
        if start_idx == -1:
            print("ERROR: APP_INITIALIZATION_STATE not found")
            return
            
        start_idx += len(start_marker)
        end_idx = html.find(end_marker, start_idx)
        if end_idx == -1:
            print("ERROR: APP_FLAGS not found after APP_INITIALIZATION_STATE")
            return
            
        json_str = html[start_idx:end_idx]
        
        # Parse it
        data = json.loads(json_str)
        
        print(f"Data is of type: {type(data)}")
        if isinstance(data, list):
            print(f"List length: {len(data)}")
            for i, item in enumerate(data):
                item_preview = str(item)[:100]
                print(f"Item {i} type: {type(item)}, preview: {item_preview}")
        
        with open('extracted_data.json', 'w', encoding='utf-8') as f:
            json.dump(data, f, indent=2)
            
        print("Success! Wrote extracted_data.json")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == '__main__':
    main()
